client/post-editor/editor-drawer/taxonomies.jsx

```diff
@@ -24,36 +25,26 @@ function EditorDrawerTaxonomies( { siteId, postType, postTerms, taxonomies } ) {
 				<QueryTaxonomies { ...{ siteId, postType } } />
 			) }
 			{ map( taxonomies, ( taxonomy ) => {
-				if ( 'post_format' === taxonomy.name ) {
+				const { name, label, hierarchical } = taxonomy;
+				if ( 'post_format' === name ) {
 					// Post format has its own dedicated accordion
 					return;
 				}
 
-				if ( taxonomy.hierarchical ) {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="category" /> }
-						>
-							Work in Progress
-						</Accordion>
-					);
-				} else {
-					return (
-						<Accordion
-							key={ taxonomy.name }
-							title={ taxonomy.label }
-							icon={ <Gridicon icon="tag" /> }
-						>
-							<TermTokenField
-								postTerms={ postTerms }
-								taxonomyName={ taxonomy.name }
-								taxonomyLabel={ taxonomy.label }
-							/>
-						</Accordion>
-					);
-				}
+				const icon = hierarchical ? 'folder' : 'tag';
+
+				return (
+					<Accordion
+						key={ name }
+						title={ label }
+						icon={ <Gridicon icon={ icon } /> }
+					>
+					{ hierarchical
+						? <TermSelector postTerms={ postTerms } taxonomyName={ name } />
+						: <TermTokenField postTerms={ postTerms } taxonomyName={ name } taxonomyLabel={ label } />
```

Comment:
Another aside @nylen , we should use `getTerm` from within `<TermTokenField />`'s `connect` selector to retrieve the label, not force it to be passed in as a prop.